### ISSUE TITLE
GitHub Organizations

### ISSUE THREAD

**Issue by GhostofGoes** [overleaf/overleaf#475]
**tl;dr: add ability to import from GitHub repositories owned by organizations.**

Currently, when you create a new project and select "Import from GitHub", you're presented with a list of the repositories owned by the linked github account. However, I can't find a way to link to repositories that are under an organization.

My specific use case is at the President for a club at my school. Our meeting notes, constitution, and code snippets are GitHub under a organization for the club. Being unable to link to repos under that org means the constitution has to be manually copied over from ShareLatex to the repo folder, then pushed to GitHub once we ratify a change.

**Comment by sindrehan** [comment]
Would be nice to see a solution to this issue, it is currently a limitation for me as well.

**Comment by devmessias** [comment]
I'm a bit confused about what features exist in the open source version. I installed the open source version and it has no integration with github.

**Comment by GhostofGoes** [comment]
@devmessias Maybe it's only in the paid version. Could I close this?

**Comment by devmessias** [comment]
@GhostofGoes , but has a some piece of code which has something about this https://github.com/sharelatex/web-sharelatex/blob/723c08706340b09a07f30c45ad75ef9e8d21fff0/app/coffee/models/User.coffee

**Comment by GhostofGoes** [comment]
Hmm, alright, I'll leave it open then

**Comment by tcampion** [comment]
I have what I think is the same issue: the list of repositories which sharelatex allows me to import from github includes repositories that I own, but not repositories which are shared with me.

To clarify, I do have the premium ShareLatex. Thus I am able to import some github repositories. But I am not able to import every github repository to which I have access. It appears that I'm allowed to import from repositories that I own, but not those which are shared with me.

Is this an intended limitation?

**Comment by etosch** [comment]
I would also like to see this feature; we use ShareLatex in our research group, which has a Github org associated with it. Right now we need to have someone be the main owner and manually port it to the associated Github repo.

**Comment by simenojensen** [comment]
We are using Overleaf in our company and would also like to see this feature implemented.

**Comment by pconrad** [comment]
Not all of the GitHub organizations that even I owned and created are showing up for me.  I don't understand the pattern of why some are showing up and others are not.

**Comment by iwishiwasaneagle** [comment]
Can confirm that this works well

> You need to remove the authorized OAuth apps for Overleaf in github configuration, then link again Overleaf to Github and grant the access to the listed organizations.

**Comment by mserranom** [comment]
Thanks for confirming @viktorivan @iwishiwasaneagle!

**Comment by jooyoungseo** [comment]
This still does not work and I was wondering if there was any inadvertent regression recently.

**Comment by dvaliant** [comment]
Hello! Same as above -this doesn't seem to be working unfortunately. I've synced my github account to overleaf (cloud) and made sure that "Organization access" was green/granted, but when I try to start a new project and import it from github, I can see only my personal repos, not the org ones. Any suggestion?

**Comment by klieret** [comment]
I can confirm the report from @dvaliant.

@mserranom Could you reopen this issue? It seems like this regressed or was never solved and many people are frustrated by this.

**Comment by mserranom** [comment]
I'm afraid we haven't been able to reproduce the problem. If you can send a complete reproduction scenario, with the exact organisation setup, etc. we can further investigate.

**Comment by jflanigan** [comment]
In the organization, under settings, go to Third-party Access "OAuth application policy". If it says "Policy: Access restricted" you can click "Remove restrictions" and then the organization will show up in the Overleaf dropdown menu when you try to link a project to github.  This worked for me.

**Comment by NorDar** [comment]
Found this via google and it led me to the correct settings in GH:
My situation:
* Linked GitHub and Overleaf before joining and/or creating organizations on GitHub
* Could only see my personal profile to save a Overleaf project to

My resolution:
* On GitHub go to *Settings* -> *Applications* -> *Authorized OAuth Apps* (https://github.com/settings/applications)
* Scroll down to *Overleaf GitHub Sync* and open it
* Scroll down to *Organization access* and click `Grant` to the organization I wanted to grant access to.

I have the role owner in the organization.
